File: src/main/java/com/sy/coladay/Bootstrap.java
package com.sy.coladay;

import static java.lang.System.getenv;

import com.sy.coladay.init.ApplicationInitializer;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * Boot loader of the cola application.
 *
 * @author selim
 */
@SpringBootApplication
public class Bootstrap {

  public static void main(String[] args) {

    if ("true".equalsIgnoreCase(getenv("ENABLE_COLADAY_INIT_MODE"))) {
      var initializer = new ApplicationInitializer();
      initializer.initApplication();
      return;
    }
    if ("true".equalsIgnoreCase(getenv("ENABLE_COLADAY_LOCAL_DEV"))) {
      var initializer = new ApplicationInitializer();
      initializer.initApplication();
      SpringApplication.run(Bootstrap.class, args);
      return;
    }
    SpringApplication.run(Bootstrap.class, args);
  }


}

File: src/main/java/com/sy/coladay/Bootstrap.kt
package com.sy.coladay

import com.sy.coladay.init.applicationInitializer
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication

/**
 * Bootloader of the cola application.
 *
 * @author selim
 */
@SpringBootApplication
open class Bootstrap {}

fun main(args: Array<String>) {
    if ("true".equals(System.getenv("ENABLE_COLADAY_INIT_MODE"), ignoreCase = true)) {
        applicationInitializer()
        return
    }
    if ("true".equals(System.getenv("ENABLE_COLADAY_LOCAL_DEV"), ignoreCase = true)) {
        applicationInitializer()
        SpringApplication.run(Bootstrap::class.java, *args)
        return
    }
    SpringApplication.run(Bootstrap::class.java, *args)
}